draft-thomson-http2-client-certs.md

```diff
@@ -185,26 +191,18 @@ client authentication.  In TLS 1.3 a server does this by sending a new TLS 1.3
 CertificateRequest.
 
 The certificate_request_id (name TBD) field of the TLS CertificateRequest is
-populated by the server with the stream identifier of the request that triggered
-the request.  That is, the value includes 32-bit value with a single reserved
-bit and the 31-bit stream identifier in network byte order.  This allows a
-client to correlate the TLS CertificateRequest with a request.
-
-The stream identifier provided in certificate_request_id MUST identify a
-client-initiated stream.  The server MUST NOT send a TLS CertificateRequest for
-a server-initiated stream or for a stream that does not have an outstanding
-request.  In other words, a server can only initiate a request for a client
-certificate from the "open" or "half-closed (remote)" stream states.
-
-A client that receives a TLS CertificateRequest with a certificate_request_id
-that does not identify a stream in a valid state ("open" or "half-closed
-(local)" for clients) SHOULD treat this as a connection error of type
-PROTOCOL_ERROR.
+populated by the server with a random value.  The server SHOULD then send a
+WAITING_FOR_AUTH frame (see {{frame}}) on any streams where the server is
+awaiting client authentication before responding.  This allows a client to
+correlate the TLS CertificateRequest with one or more outstanding requests.
```

Comment:
I think that this needs to be a MUST send WAITING_FOR_AUTH and that it has to come before the CertificateRequest, for the reasons that Andrei discussed.